### ISSUE TITLE
Reimplementation of Performance in the Paper's Results Table

### ISSUE THREAD

**Issue by hanoong7** [VLKEB/VLKEB#4]
Hello,

I found the VLKEB paper fascinating and am currently working on reconstructing the experiments presented in it. I downloaded the provided pretrain model mend.pt, and following the instructions in the Readme, I used facebook/opt-2.7b and google-bert/bert-base-uncased, along with downloading blip2_pretrained_opt2.7b.pth and hugging_cache/eva_vit_g.pth into the hugging_cache directory.

However, when I ran test_MEND_Blip2OPT in mutimodal_edit.py, the image_loc value I obtained was 79.08, which is significantly different from the 89.05 reported in the paper.

Is there something I might have done incorrectly? Below are the results of running test_MEND_Blip2OPT and the content of hparams/MEND/blip2.yaml.

Additionally, could you please confirm if, in the sequential edit branch, the only additions to MultimodalTrainer.py compared to the main branch are test_sequencial, test_sequencial_step, and _inline_seq_log? Are there any other changes in the code or functions, such as in BaseTrainer.py?

# Result:
  loss/edit_val       :  0.06862
  loss/image_edit_val :  0.06257
  loss/loc_val        :  0.00027
  edit/acc_val        :  0.98844
  edit/log_prob_val   : -0.06862
  edit/prob_val       :  0.98017
  inner/acc_val       :  0.98926
  image_rephrase/acc_val:  0.98798
  time/edit_val       :  0.87529
  loc/acc_val         :  0.99359
  image_loc/acc_val   :  0.79084
  loss/total_val      :  0.65759
  loss/total_edit_val :  0.65759
  memory/alloc_max_val:  0.00000
  memory/res_max_val  :  0.00000
  port/acc_val        :  0.28796
  eval_time/elapsed   :  3254.99364
  eval_time/average   :  2.54694

  # Model
  device: 0
  alg_name: "MEND"
  name: facebook/opt-2.7b
  model_name: blip2
  model_class: Blip2OPT
  tokenizer_class: GPT2Tokenizer
  tokenizer_name: facebook/opt-2.7b
  inner_params:
  - opt_model.model.decoder.layers.29.fc1.weight
  - opt_model.model.decoder.layers.29.fc2.weight
  - opt_model.model.decoder.layers.30.fc1.weight
  - opt_model.model.decoder.layers.30.fc2.weight
  - opt_model.model.decoder.layers.31.fc1.weight
  - opt_model.model.decoder.layers.31.fc2.weight
  
  # Method
  alg: MEND
  lr: 1e-6
  edit_lr: 1e-4
  lr_lr: 1e-4
  lr_scale: 1.0
  seed: 42
  cedit: 0.1
  iedit: 0.1
  cloc: 1.0
  cbase: 1.0
  dropout: 0.0
  train_base: False
  no_grad_layers: null
  one_sided: False
  n_hidden: 1
  hidden_dim: null
  init: id
  norm: True
  combine: True
  x_only: False
  delta_only: False
  act: relu
  rank: 1920
  mlp_class: IDMLP
  shared: True
  archive: VLKEB-models/mend/mend_blip2.pt
  
  # Train
  batch_size: 1
  model_save_pt: 5000
  silent: False
  #max_epochs: 1
  max_iters: 40000
  log_interval: 100
  eval_log_interval: 1000
  final_eval: True
  val_interval: 5000
  early_stop_patience: 20000
  early_stop_key: "loss/total_edit_val"
  eval_only: True
  half: False
  debug: False
  save: False
  verbose: True
  
  val_batch_size: 1
  accumulate_bs: 2
  val_steps: 500 # only for debug
  opt: Adam
  grad_clip: 100.
  
  # Output
  results_dir: ./results
  
  # Multimodal
  qformer_checkpoint: hugging_cache/blip2_pretrained_opt2.7b.pth
  qformer_name_or_path: google-bert/bert-base-uncased
  state_dict_file: hugging_cache/eva_vit_g.pth
  
  # image
  coco_image: datasets/VLKEB_images/mmkb_images
  rephrase_image: datasets/VLKEB_images/mmkb_images

**Comment by hanhuang22** [comment]
Hello, sorry for the late reply. I have checked the config you provide and that is same as mine. I checked the step of my checkpoint, and it is 35000. I wonder if this makes the difference. 

And regarding BaseTrainer.py in sequential edit branch, there are just minor changes in results_path. You may check the difference with git command like: 
```git
git diff main multihop_and_sequential easyeditor/trainer/BaseTrainer.py
```

**Comment by hanoong7** [comment]
Thank you for your response.

However, the value of steps in my checkpoint is also 35,000, and I still cannot verify the results of the image_loc paper.
Is there any other method I can try to validate the paper's results?

**Comment by hanhuang22** [comment]
I may have misunderstood your question that I thought you train mend method on BLIP2 on your own. Since you use my pretrained model, I test it today and find that results same as yours.
To solve this issue, I inspected my local git log and find the change of data processing of image locality results in this discrepancy.
To get the results in the paper, make adjustments to this part of [dataset](https://github.com/VLKEB/VLKEB/blob/7c8d187c2831211dab44344ad02883a3d89ac5f3/easyeditor/dataset/coco_caption.py#L204):
```python
        # m_loc
        loc_image = {}
        loc_image['image'] = torch.stack(m_loc_image, dim=0)
        # loc_image['text_input'] = [" ".join([q, a]) for q, a in zip(m_loc_q, m_loc_a)]
        # loc_image['prompts_len'] = [len(self.tok.encode(q, add_special_tokens=False)) for q in m_loc_q]
        loc_image['text_input'] = [self.prompt.format(q) + a for q, a in zip(m_loc_q, m_loc_a)]
        loc_image['prompts_len'] = [len(self.tok.encode(self.prompt.format(q), add_special_tokens=False)) for q in m_loc_q]
        loc_image['labels'] = self.tok(m_loc_a, add_special_tokens=False, return_tensors="pt",)["input_ids"]
```
Thank you very much for bringing up this issue, we would revise our code or paper later.

**Comment by hanoong7** [comment]
Thank you for your response! 
Now, I am able to achieve the same results as those in the experiment of the paper.
